bin/lowly.js

```diff
@@ -26,13 +30,26 @@ const cli = meow(`
 
 const input = cli.input
 
-function getOutputFileName(input) {
+const getOutputFileName = input => {
   const inputPath = path.parse(input)
   return path.resolve(inputPath.dir, inputPath.name + '-lowly' + inputPath.ext)
 }
 
+const checkType = (buff, file) => {
+  return new Promise((resolve, reject) => {
+    magic.detect(buff, (err, res) => {
+      if (res.indexOf('image') >= 0) {
+        resolve(buff)
+      } else {
+        reject(err || 'Ignored file : ' + file + ' (not an image)')
+      }
+    })
+  })
+}
+
 input.forEach(file => {
   readFile(path.resolve(file))
+    .then(buff => checkType(buff, file))
```

Comment:
Is it possible to verify the files before read them? I'm sure but this can reduce the processing cost.